transmogrifier/sources/xml/ead.py

```diff
@@ -425,6 +374,149 @@ def get_content_type(cls, source_record: Tag) -> list[str] | None:
+        """Generate a full name identifier URL with the specified scheme.
+
+        Args:
+            name_element: A BeautifulSoup Tag representing an EAD
+                name XML field.
+        """
+        if identifier := name_element.get("authfilenumber"):
+            source = name_element.get("source")
+            if source in ["lcnaf", "naf"]:
+                base_url = "https://lccn.loc.gov/"
+            elif source == "snac":
+                base_url = "https://snaccooperative.org/view/"
+            elif source == "viaf":
+                base_url = "http://viaf.org/viaf/"
+            else:
+                base_url = ""
+            return [base_url + identifier]
+        return None
+
+    @classmethod
+    def get_dates(cls, source_record: Tag) -> list[timdex.Date] | None:
+        dates = []
+        collection_description_did = cls._get_collection_description_did(source_record)
+        source_record_id = cls.get_source_record_id(source_record)
+        dates.extend(
+            cls._parse_date_elements(collection_description_did, source_record_id)
+        )
+        return dates or None
+
+    @classmethod
+    def _parse_date_elements(
+        cls, collection_description_did: Tag, source_record_id: str
+    ) -> list[timdex.Date]:
+        """
+        Dates are derived from <archdesc><unitdate> elements with the @normal attribute.
+
+        The method will iterate over these elements, validating The values assigned to
+        the @normal attribute. If the date value passes validation, a timdex.Date
+        instance is created and added to the list of valid timdex.Dates that is
+        returned.
+
+        Note: If the date values in a provided date range are the same, a
+          single date is retrieved.
+        """
+        dates = []
+        for date_element in collection_description_did.find_all("unitdate", normal=True):
+            date_value = date_element.get("normal", "").strip()
+            if date_value == "":
+                continue
+
+            date_instance = timdex.Date()
+
+            # get valid date ranges or dates
+            if "/" in date_value:
+                gte_date, lte_date = date_value.split("/")
+                if gte_date != lte_date:
+                    if validate_date_range(
+                        gte_date,
+                        lte_date,
+                        source_record_id,
+                    ):
+                        date_instance.range = timdex.DateRange(
+                            gte=gte_date,
+                            lte=lte_date,
+                        )
+                else:
+                    # get valid date (if dates in ranges are the same)
+                    date_string = gte_date
+                    if validate_date(
+                        date_string,
+                        source_record_id,
+                    ):
+                        date_instance.value = date_string
```

Comment:
This should be broken out as a separate method to improve readability, `Datacite` has a `_parse_date_range` method that would be very similar

Reply:
Please review the [changes in latest commit](https://github.com/MITLibraries/transmogrifier/pull/187/commits/25b47059e3994073e3a4d76e2b33cbf6e62b3fd8#diff-abec0db581c2be5b547c40215c27fb590a04d123795a49fc22eb674ec5ac50d1R487-R509)!